### DISCUSSION TITLE
Explications de ton demo.py

### DISCUSSION THREAD

**Discussion by nmathey** [MadeInPierre/finalynx#65] [Q&A] [answered]
Hello,

Dans ton demo.py quelle est la logique derrière tes folders "Daily" et "Monthly" ? 

Côté "bucket_garanti", il s'agit en sorte d'un regroupement de plusieurs enveloppes que l'on souhaite ensuite "dispatcher" entre différents objectifs?
Devons nous lire dans l'exemple ainsi:
1. A court terme je souhaite atteindre un filet de sécurité de 6000 et une enveloppe de 2000 pour des projets/voyages soit un bucket de minimum 8000
2. A moyen terme je souhaite atteindre une enveloppe de 20000: en plus des 8000 soit un bucket total de minimum 28000 ou est-ce le bucket qui doit être à 20000?
3. A long terme je souhaite que mon pot commun "garanti" représente 25% de mon patrimoine financier incluant les objectifs courts et moyens termes pré-cité?

Merci d'avance :)

**Comment by MadeInPierre** [comment, marked as answer]
Hello ! Merci de ton intérêt envers ce projet assez fun pour moi ^^ 

Alors, je commence par une [petite présentation](https://finalynx.readthedocs.io/en/latest/tutorials/getting_started.html#define-your-portfolio) du concept. Le but est de déclarer notre portefeuille selon une structure de dossiers comme sur Windows Explorer classique, avec plusieurs options:
- La classe `Portfolio("Mon Portefeuille", children=[...])` est la racine du portefeuille, tous les éléments iront dans la liste `children`.
- On peut ensuite mettre 3 types d'éléments :
    - `Line` représente chaque investissement (e.g. un ETF, une crypto, ...)
    - `Folder` est juste un ensemble de lignes, chaque Folder accepte une liste de `Line`, `Folder` ou `SharedFolder`
    - `SharedFolder` utilise un `Bucket` déclaré en avance. Chaque SharedFolder utilisant un même Bucket va prélever un montant cible `target_amount` du bucket et laisser le reste aux prochains SharedFolder.

Ainsi, la démo pourrait se résumer ainsi :
- Je déclare un `Bucket` contenant tous mes livrets & fonds euro, qui au final agissent comme une Line et serviront à plusieurs choses dans mon portefeuille.
- Daily (type `Folder`) a un CCP (type `Line`) pour les dépenses quotidiennes, Monthly c'est pareil avec mes banques traditionnelles
- "Safety net" est un `SharedFolder` qui va prélever 6000€ du bucket et agira ensuite comme un `Folder` (il affiche les lignes qu'il a utilisées pour prélever ces 6000€). Le bucket a maintenant 6000€ de moins qu'avant.
- "Voyages" prélève ensuite 2000€ du bucket
- "Medium Term" prélève 20k€. S'il n'y a pas assez, il affiche ce qu'il reste (ou est vide s'il n'y a plus rien)


Au plaisir d'échanger ! N'hésite pas pour toutes questions/propositions/idées autour de ce projet :)

**Reply by nmathey** [reply]
Parfait je comprend mieux merci pour ses précisions!
